src/igd.cpp

```diff
@@ -370,12 +370,12 @@ result<std::vector<igd>> igd::discover(net::executor exec, net::yield_context yi
         }
     }
 
-    return std::move(igds);
+    return igds;
```

Comment:
This code also removes the warning but keeps the `std::move`. Can you see if this fixes it on yours? If not, I'll poke around tonight to see if I can get gcc 7.5

```suggestion
    return {std::move(igds)};
```

Reply:
Yes, that compiles for me as well.